### PR TITLE
Adapt to abstract tests in the TCK

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes https://github.com/spring-cloud/spring-cloud-deployer-mesos/issues/30

Requires https://github.com/spring-cloud/spring-cloud-deployer/pull/111

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java [hunk 1]
```diff
@@ -229,6 +229,7 @@ protected TaskStatus buildTaskStatus(ChronosTaskLauncherProperties properties, S
 			return new TaskStatus(id, LaunchState.complete, new HashMap<>());
 		}
 		else {
+			// TODO: state == idle could indicate cancelled?
```
Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java:232]: @trisberg You may want to have a look at this

File: src/test/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncherIntegrationTests.java [hunk 2]
```diff
@@ -37,173 +27,68 @@
...
-		Assert.assertThat(deploymentId, eventually(hasStatusThat(
-				Matchers.<TaskStatus>hasProperty("state", Matchers.is(complete))), timeout.maxAttempts, timeout.pause));
-
-		((ChronosTaskLauncher)getTargetObject(taskLauncher)).cleanup(deploymentId);
-	}
-
-	protected String randomName() {
-		return "job-" + UUID.randomUUID().toString().substring(0, 18);
+	@Override
+	@Ignore("Currently reported as failed instead of cancelled")
```
Comment (ericbottard) [inline src/test/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncherIntegrationTests.java:72]: @trisberg Remove once https://github.com/spring-cloud/spring-cloud-deployer-mesos/issues/32 is fixed

File: src/test/java/org/springframework/cloud/deployer/spi/mesos/marathon/MesosAppDeployerIntegrationTests.java [hunk 3]
```diff
@@ -47,6 +50,18 @@ protected AppDeployer appDeployer() {
 	}
 
 	@Override
+	protected String randomName() {
+		return super.randomName().toLowerCase(); // Marathon wants lower-case ids
+	}
+
+	@Test
+	@Override
+	@Ignore("Needs to be implemented")
```
Comment (ericbottard) [inline src/test/java/org/springframework/cloud/deployer/spi/mesos/marathon/MesosAppDeployerIntegrationTests.java:59]: @trisberg This is not used in SCDF, but is theoretically in the deployer contract

**Comment by trisberg** [comment]
Squashed and merged as commit 47f255014dd8f3a3d641082f90fc1842d2b83d1d

**Pull request closed without merging**